base/math.jl

```diff
@@ -696,7 +696,14 @@ function _hypot(x, y)
     end
     return h*scale*oneunit(axu)
 end
-_hypot(x::Float16, y::Float16) = Float16(_hypot(Float32(x), Float32(y)))
+@inline function _hypot(x::Float32, y::Float32)
+    _x, _y = Float64(x), Float64(y)
+    return Float32(sqrt(@fastmath _x*_x + _y*_y))
```

Comment:
```suggestion
    if isinf(_x) || isinf(_y)
        return Inf32
    end
    return Float32(sqrt(@fastmath _x*_x + _y*_y))
```

Reply:
```julia
julia> function _hypot(x::Float32, y::Float32)
           _x, _y = Float64(x), Float64(y)
           return Float32(sqrt(@fastmath _x*_x + _y*_y))
       end
_hypot (generic function with 2 methods)

julia> _hypot(Float32(Inf32), 0.3f0)
Inf32
```

this already works, do we need this for performance reason?

Reply:
https://github.com/JuliaLang/julia/blob/f2d03be48d4f6748a2d9ce87cacbb7b9ab6151ed/test/math.jl#L207

this seems to test Inf s?

Reply:
From the failing tests:
```julia
Error in testset math:
Test Failed at /buildworker/worker/tester_linux64/build/share/julia/test/math.jl:286
  Expression: hypot(T(Inf), T(NaN)) === T(Inf)
   Evaluated: NaN32 === Inf32
Error in testset math:
Test Failed at /buildworker/worker/tester_linux64/build/share/julia/test/math.jl:1243
  Expression: #= /buildworker/worker/tester_linux64/build/share/julia/test/math.jl:1243 =# @inferred(hypot(T(Inf), T(NaN))) == T(Inf)
   Evaluated: NaN32 == Inf32
```